datagen/utils/btcstaking.go

```diff
@@ -79,13 +155,31 @@ func GenBTCDelegation(dir string) {
 
 	btcDelBytes, err := btcDel.Marshal()
 	require.NoError(t, err)
-	btcDelPath := filepath.Join(dir, BTC_DEL_FILENAME)
+
+	// filename
+	fpIdxListStr := make([]string, len(fpIdxList))
+	for i, idx := range fpIdxList {
+		fpIdxListStr[i] = fmt.Sprintf("%d", idx)
+	}
+	fileName := fmt.Sprintf(BTC_DEL_FILENAME, idx, strings.Join(fpIdxListStr, ","))
+
+	btcDelPath := filepath.Join(dir, fileName)
 	err = os.WriteFile(btcDelPath, btcDelBytes, 0644)
 	require.NoError(t, err)
+}
 
-	paramsBytes, err := bsParams.Marshal()
-	require.NoError(t, err)
-	paramsPath := filepath.Join(dir, BTCSTAKING_PARAMS_FILENAME)
-	err = os.WriteFile(paramsPath, paramsBytes, 0644)
-	require.NoError(t, err)
+func GenBTCDelegationsAndParams(dir string) {
+	covenantSKs, covenantQuorum := GenParams(dir)
+	// 3 FPs
+	GenFinalityProviders(dir, 3)
+	// 2 BTC delegations under 1st FPs
+	GenBTCDelegations(dir, covenantSKs, covenantQuorum, 1, []int{1})
+	GenBTCDelegations(dir, covenantSKs, covenantQuorum, 2, []int{1})
+	GenBTCDelegations(dir, covenantSKs, covenantQuorum, 3, []int{1})
+	// 2 BTC delegations under 2nd FPs
```

Comment:
```suggestion
	// 3 BTC delegations under 2nd FPs
```
?

Can the BTC del IDs overlap across different FPs?

Reply:
It won't. BTC delegation ID is the staking tx hash, and the staking tx is certainly unique